torch/csrc/jit/codegen/cuda/predicate_compute.cpp

```diff
@@ -131,20 +154,6 @@ kir::Bool* PredicateCompute::getInlinePredicate(
   auto root_indices = pred_inds.first;
   bool use_maybe_rfactor = pred_inds.second;
 
-  if (out_tv->getMemoryType() == MemoryType::Local && out_tv->hasReduction() &&
-      !use_maybe_rfactor) {
-    auto tv_filter_inp_view =
-        ir_utils::filterByType<TensorView>(expr->inputs());
-    auto has_tv_inputs = tv_filter_inp_view.begin() != tv_filter_inp_view.end();
-    // If predicates doesn't need maybe_rfactor, but it has reduction axes, and
-    // expr has no inputs, we're pretty confident we're intializing a reduction
-    // buffer. If we're initing a reduction buffer don't generate an inline
-    // predicate.
-    if (!has_tv_inputs) {
-      return ir_builder.create<kir::Bool>(true);
-    }
-  }
```

Comment:
Why is this part removed?

Reply:
It seems accidental, good catch!

Reply:
Actually, it may not have been so accidental - adding it back results in 22 failures.

I'm trying to track back the history of the change. It's been a while and my memory is failing me :(